plugins/woocommerce/includes/gateways/paypal/includes/class-wc-gateway-paypal-request.php

```diff
@@ -322,24 +322,45 @@ public function authorize_or_capture_payment( $order, $action_url, $action = WC_
 	 * @throws Exception If the PayPal payment capture fails.
 	 */
 	public function capture_authorized_payment( $order ) {
-		if ( ! $order || ! $order->get_transaction_id() ) {
-			WC_Gateway_Paypal::log( 'PayPal authorization ID not found. Cannot capture payment.' );
+		if ( ! $order ) {
+			WC_Gateway_Paypal::log( 'Order not found to capture authorized payment.' );
 			return;
 		}
 
+		$paypal_order_id = $order->get_meta( '_paypal_order_id', true );
+		// Skip if the PayPal Order ID is not found. This means the order was not created via the Orders v2 API.
+		if ( ! $paypal_order_id ) {
+			WC_Gateway_Paypal::log( 'PayPal Order ID not found to capture authorized payment. Order ID: ' . $order->get_id() );
+			return;
+		}
+
+		$capture_id = $order->get_meta( '_paypal_capture_id', true );
 		// Skip if the payment is already captured.
+		if ( $capture_id ) {
+			WC_Gateway_Paypal::log( 'PayPal payment is already captured. PayPal capture ID: ' . $capture_id . '. Order ID: ' . $order->get_id() );
```

Comment:
Nit: almost certainly for a follow-up PR, but it would be helpful if we could pass in additional details to `log()`. In this case, the capture and order IDs could go in that data.